sql/log_event.cc

```diff
@@ -10723,7 +10723,21 @@ int Rows_log_event::do_apply_event(Relay_log_info const *rli)
       DBUG_RETURN(-1);
     }
     else if (state == GTID_STATEMENT_SKIP)
+    {
```

Comment:
Do we need similar cleanup logic for the previous branch
`if (state == GTID_STATEMENT_CANCEL)`?